case_studies/CARD/Bug-Drug Code.R

```diff
@@ -0,0 +1,257 @@
+    filter(grepl(bug, Pathogen_Full_Name, ignore.case = TRUE))
+
+  # Optionally exclude multiclass resistance if exclude_multiclass = TRUE
+  if (exclude_multiclass) {
+    filtered_data <- filtered_data %>%
+      filter(!grepl(";", Drug))  # Only include entries with single drug classes
+  }
+
+  # Optionally restrict to species-specific mechanisms if species_restricted = TRUE
+  if (species_restricted) {
+    filtered_data <- filtered_data %>%
+      filter(Pathogen_Full_Name == bug)  # Include only entries with exact match to the bug of interest
+  }
+
+  return(filtered_data)
+}
+
+# Usage example for Staphylococcus aureus resistant to DAP, including multispecies and multiclass resistance
+filtered_data_saurdap <- filter_resistance_mechanisms(
+  data = merged_data_pathogens,
+  drug = "DAP",
+  bug = "Staphylococcus aureus",
+  exclude_multiclass = FALSE,
+  species_restricted = FALSE
+)
+
+# View the filtered results
+View(filtered_data_saurdap)
+
+
+# Fetch FASTA sequence from Entrez
+fetch_fasta_sequence <- function(protein_accession) {
+  tryCatch({
+    # Fetch the FASTA sequence using Entrez
+    fasta_seq <- rentrez::entrez_fetch(db = "protein",
+                                       id = protein_accession,
+                                       rettype = "fasta",
+                                       retmode = "text")
+    
+    if (!is.null(fasta_seq)) {
+      # Ensure the first line starts with ">"
+      if (!grepl("^>", fasta_seq[1])) {
+        fasta_seq[1] <- paste0(">", fasta_seq[1])
+      }
+      
+      # Split the sequence into lines
+      lines <- str_split(fasta_seq, "\n")[[1]]
+      
+      # Join the lines back together
+      fasta_seq <- paste(lines, collapse = "\n")
+      
+      return(fasta_seq)
+    } else {
+      warning(paste("Failed to retrieve FASTA sequence for protein accession:", protein_accession))
+      return(NULL)
+    }
+  }, error = function(e) {
+    warning(paste("Error fetching FASTA sequence for protein accession:", protein_accession, ":", e$message))
+    return(NULL)
+  })
+}
+
+
+# Define the output file for the FASTA sequences
+output_fasta_file <- "Staph_aureus_Daptomycin_sequences.fasta"
+
+# Initialize an empty character vector to store the sequences
+combined_sequences <- character()
+
+# Loop through each Protein Accession in the filtered data to fetch sequences
+for (i in 1:nrow(filtered_data_saurdap)) {
+  # Get the Protein Accession ID
+  Protein_accession <- filtered_data_saurdap$Protein_Accession[i]
+  
+  cat("Fetching sequence for Protein Accession:", protein_accession, "\n")  # Debugging message
+  
+  # Fetch the FASTA sequence
+  fasta_sequence <- fetch_fasta_sequence(protein_accession)
+  
+  # If the sequence was fetched successfully, add it to the combined_sequences vector
+  if (!is.null(fasta_sequence)) {
+    combined_sequences <- c(combined_sequences, fasta_sequence)
+    cat("Successfully fetched sequence for:", protein_accession, "\n")
```

Comment:
Not sure if this is for multiple or single accession numbers. change accordingly?
```suggestion
    cat("Successfully fetched sequences for:", protein_accession, "\n")
```